### ISSUE TITLE
How can I convert a trained pth file to diffusers

### ISSUE THREAD

**Issue by Victcode** [Alpha-VLLM/Lumina-Image-2.0#47]

**Comment by Victcode** [comment]
I modified the code in convert_lumina_to_diffusers.py, it can convert lumina 2.0.I post the code here if anyone needs it
`mport argparse
import os

import torch
from safetensors.torch import load_file
from diffusers import Lumina2Transformer2DModel,Lumina2Pipeline


def main(args):
    # all_sd = load_file(args.origin_ckpt_path, device="cpu")
    all_sd = torch.load(args.origin_ckpt_path, weights_only=False)

    converted_state_dict = {}
    # patch embed
    converted_state_dict["x_embedder.weight"] = all_sd["x_embedder.weight"]
    converted_state_dict["x_embedder.bias"] = all_sd["x_embedder.bias"]

    # time and caption embed
    converted_state_dict["time_caption_embed.timestep_embedder.linear_1.weight"] = all_sd["t_embedder.mlp.0.weight"]
    converted_state_dict["time_caption_embed.timestep_embedder.linear_1.bias"] = all_sd["t_embedder.mlp.0.bias"]
    converted_state_dict["time_caption_embed.timestep_embedder.linear_2.weight"] = all_sd["t_embedder.mlp.2.weight"]
    converted_state_dict["time_caption_embed.timestep_embedder.linear_2.bias"] = all_sd["t_embedder.mlp.2.bias"]
    converted_state_dict["time_caption_embed.caption_embedder.0.weight"] = all_sd["cap_embedder.0.weight"]
    converted_state_dict["time_caption_embed.caption_embedder.1.weight"] = all_sd["cap_embedder.1.weight"]
    converted_state_dict["time_caption_embed.caption_embedder.1.bias"] = all_sd["cap_embedder.1.bias"]

    for i in range(2):
        ## noise_refiner
        converted_state_dict[f"noise_refiner.{i}.attn.norm_q.weight"] = all_sd[f"noise_refiner.{i}.attention.q_norm.weight"]
        converted_state_dict[f"noise_refiner.{i}.attn.norm_k.weight"] = all_sd[f"noise_refiner.{i}.attention.k_norm.weight"]
        converted_state_dict[f"noise_refiner.{i}.attn.to_q.weight"] = all_sd[f"noise_refiner.{i}.attention.qkv.weight"][:2304, :]
        converted_state_dict[f"noise_refiner.{i}.attn.to_k.weight"] = all_sd[f"noise_refiner.{i}.attention.qkv.weight"][2304:3072, :]
        converted_state_dict[f"noise_refiner.{i}.attn.to_v.weight"] = all_sd[f"noise_refiner.{i}.attention.qkv.weight"][3072:, :]
        converted_state_dict[f"noise_refiner.{i}.attn.to_out.0.weight"] = all_sd[f"noise_refiner.{i}.attention.out.weight"]

        converted_state_dict[f"noise_refiner.{i}.feed_forward.linear_1.weight"] = all_sd[f"noise_refiner.{i}.feed_forward.w1.weight"]
        converted_state_dict[f"noise_refiner.{i}.feed_forward.linear_2.weight"] = all_sd[f"noise_refiner.{i}.feed_forward.w2.weight"]
        converted_state_dict[f"noise_refiner.{i}.feed_forward.linear_3.weight"] = all_sd[f"noise_refiner.{i}.feed_forward.w3.weight"]

        converted_state_dict[f"noise_refiner.{i}.ffn_norm1.weight"] = all_sd[f"noise_refiner.{i}.ffn_norm1.weight"]
        converted_state_dict[f"noise_refiner.{i}.ffn_norm2.weight"] = all_sd[f"noise_refiner.{i}.ffn_norm2.weight"]
        
        converted_state_dict[f"noise_refiner.{i}.norm1.norm.weight"] = all_sd[f"noise_refiner.{i}.attention_norm1.weight"]
        converted_state_dict[f"noise_refiner.{i}.norm1.linear.weight"] = all_sd[f"noise_refiner.{i}.adaLN_modulation.1.weight"]
        converted_state_dict[f"noise_refiner.{i}.norm1.linear.bias"] = all_sd[f"noise_refiner.{i}.adaLN_modulation.1.bias"]
        converted_state_dict[f"noise_refiner.{i}.norm2.weight"] = all_sd[f"noise_refiner.{i}.attention_norm2.weight"]


        ## context_refiner
        converted_state_dict[f"context_refiner.{i}.attn.norm_q.weight"] = all_sd[f"context_refiner.{i}.attention.q_norm.weight"]
        converted_state_dict[f"context_refiner.{i}.attn.norm_k.weight"] = all_sd[f"context_refiner.{i}.attention.k_norm.weight"]
        converted_state_dict[f"context_refiner.{i}.attn.to_q.weight"] = all_sd[f"context_refiner.{i}.attention.qkv.weight"][:2304, :]
        converted_state_dict[f"context_refiner.{i}.attn.to_k.weight"] = all_sd[f"context_refiner.{i}.attention.qkv.weight"][2304:3072, :]
        converted_state_dict[f"context_refiner.{i}.attn.to_v.weight"] = all_sd[f"context_refiner.{i}.attention.qkv.weight"][3072:, :]
        converted_state_dict[f"context_refiner.{i}.attn.to_out.0.weight"] = all_sd[f"context_refiner.{i}.attention.out.weight"]

        converted_state_dict[f"context_refiner.{i}.feed_forward.linear_1.weight"] = all_sd[f"context_refiner.{i}.feed_forward.w1.weight"]
        converted_state_dict[f"context_refiner.{i}.feed_forward.linear_2.weight"] = all_sd[f"context_refiner.{i}.feed_forward.w2.weight"]
        converted_state_dict[f"context_refiner.{i}.feed_forward.linear_3.weight"] = all_sd[f"context_refiner.{i}.feed_forward.w3.weight"]

        converted_state_dict[f"context_refiner.{i}.ffn_norm1.weight"] = all_sd[f"context_refiner.{i}.ffn_norm1.weight"]
        converted_state_dict[f"context_refiner.{i}.ffn_norm2.weight"] = all_sd[f"context_refiner.{i}.ffn_norm2.weight"]
        
        converted_state_dict[f"context_refiner.{i}.norm1.weight"] = all_sd[f"context_refiner.{i}.attention_norm1.weight"] 
        converted_state_dict[f"context_refiner.{i}.norm2.weight"] = all_sd[f"context_refiner.{i}.attention_norm2.weight"]


    for i in range(26):
        # attention
        converted_state_dict[f"layers.{i}.attn.to_q.weight"] = all_sd[f"layers.{i}.attention.qkv.weight"][:2304, :]
        converted_state_dict[f"layers.{i}.attn.to_k.weight"] = all_sd[f"layers.{i}.attention.qkv.weight"][2304:3072, :]
        converted_state_dict[f"layers.{i}.attn.to_v.weight"] = all_sd[f"layers.{i}.attention.qkv.weight"][3072:, :]
        converted_state_dict[f"layers.{i}.attn.to_out.0.weight"] = all_sd[f"layers.{i}.attention.out.weight"]

        # qk norm
        converted_state_dict[f"layers.{i}.attn.norm_q.weight"] = all_sd[f"layers.{i}.attention.q_norm.weight"]
        converted_state_dict[f"layers.{i}.attn.norm_k.weight"] = all_sd[f"layers.{i}.attention.k_norm.weight"]

        # feed forward
        converted_state_dict[f"layers.{i}.feed_forward.linear_1.weight"] = all_sd[f"layers.{i}.feed_forward.w1.weight"]
        converted_state_dict[f"layers.{i}.feed_forward.linear_2.weight"] = all_sd[f"layers.{i}.feed_forward.w2.weight"]
        converted_state_dict[f"layers.{i}.feed_forward.linear_3.weight"] = all_sd[f"layers.{i}.feed_forward.w3.weight"]

        # feed forward norm
        converted_state_dict[f"layers.{i}.ffn_norm1.weight"] = all_sd[f"layers.{i}.ffn_norm1.weight"]
        converted_state_dict[f"layers.{i}.ffn_norm2.weight"] = all_sd[f"layers.{i}.ffn_norm2.weight"]

        converted_state_dict[f"layers.{i}.norm1.norm.weight"] = all_sd[f"layers.{i}.attention_norm1.weight"]
        converted_state_dict[f"layers.{i}.norm1.linear.weight"] = all_sd[f"layers.{i}.adaLN_modulation.1.weight"]
        converted_state_dict[f"layers.{i}.norm1.linear.bias"] = all_sd[f"layers.{i}.adaLN_modulation.1.bias"]
        converted_state_dict[f"layers.{i}.norm2.weight"] = all_sd[f"layers.{i}.attention_norm2.weight"]

    # final layer
    # converted_state_dict["norm_out.norm.weight"] = all_sd["norm_final.weight"]
    converted_state_dict["norm_out.linear_1.weight"] = all_sd["final_layer.adaLN_modulation.1.weight"]
    converted_state_dict["norm_out.linear_1.bias"] = all_sd["final_layer.adaLN_modulation.1.bias"]
    converted_state_dict["norm_out.linear_2.weight"] = all_sd["final_layer.linear.weight"]
    converted_state_dict["norm_out.linear_2.bias"] = all_sd["final_layer.linear.bias"]
    

    # Lumina-2.0 2.6B
    transformer = Lumina2Transformer2DModel(
        sample_size=128,
        patch_size=2,
        in_channels=16,
        hidden_size=2304,
        num_layers=26,
        num_attention_heads=24,
        num_kv_heads=8,
        multiple_of=256,
        ffn_dim_multiplier=None,
        norm_eps=1e-5,
        scaling_factor=1.0,
        cap_feat_dim=2304,
        num_refiner_layers=2,
        axes_dim_rope=(32, 32, 32),
        axes_lens=(300, 512, 512),
    )
    transformer.load_state_dict(converted_state_dict, strict=True)

    num_model_params = sum(p.numel() for p in transformer.parameters())
    print(f"Total number of transformer parameters: {num_model_params}")

    if args.only_transformer:
        transformer.save_pretrained(os.path.join(args.dump_path, "transformer"))
    # else:
    #     scheduler = FlowMatchEulerDiscreteScheduler()

    #     vae = AutoencoderKL.from_pretrained("stabilityai/sdxl-vae", torch_dtype=torch.float32)

    #     tokenizer = AutoTokenizer.from_pretrained("google/gemma-2b")
    #     text_encoder = AutoModel.from_pretrained("google/gemma-2b")

    #     pipeline = LuminaPipeline(
    #         tokenizer=tokenizer, text_encoder=text_encoder, transformer=transformer, vae=vae, scheduler=scheduler
    #     )
    #     pipeline.save_pretrained(args.dump_path)


if __name__ == "__main__":
    parser = argparse.ArgumentParser()

    parser.add_argument(
        "--origin_ckpt_path", default=None, type=str, required=False, help="Path to the checkpoint to convert."
    )
    parser.add_argument(
        "--image_size",
        default=1024,
        type=int,
        choices=[256, 512, 1024],
        required=False,
        help="Image size of pretrained model, either 512 or 1024.",
    )
    parser.add_argument("--dump_path", default=None, type=str, required=True, help="Path to the output pipeline.")
    parser.add_argument("--only_transformer", default=True, type=bool, required=True)

    args = parser.parse_args()
    main(args)
`
